### PR TITLE
Choose local

### PR DESCRIPTION
**Pull request by stoic-plus**
## Choosing a Location

Picking a location on popup `choose location` - send POST req to `EvSearchController`
* Lat and Lon and the chosen's location popup HTML is saved to session 

JS
---
`chooseLocal` onClick function for `choose location` 
* Click event has access to the current url and also the lat and long through a custom data attribute on the `a` tag
* `popup_html` is accessed through being wrapped in a div -> get `innerHTML` for that div
  * Those three pieces of data are sent in POST

After that - follow the URL on the response (`/ev_map`)

---

__EvMap Show__

* Placing the chosen location marker
* Placing the station marker by iterating through the return from service

### CODE REVIEW

File: app/views/ev_map/show.html.erb [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
 
 <script>
-  var mymap = L.map('mapid').setView([<%= @ev_search_lat %>, <%= @ev_search_lon %>], 13);
+  const mymap = L.map('mapid').setView([<%= @ev_search_lat %>, <%= @ev_search_lon %>], 13);
 
   L.tileLayer('https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}', {
     attribution: 'Map data &copy; <a href="https://www.openstreetmap.org/">OpenStreetMap</a> contributors, <a href="https://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, Imagery © <a href="https://www.mapbox.com/">Mapbox</a>',
     maxZoom: 20,
     id: 'mapbox.streets',
     accessToken: 'pk.eyJ1IjoibWFyeWFncmFjZSIsImEiOiJjanM0eG9qMmIwOXlhNDVuaTNuOXMzcXd1In0.Y3D7BRb8btkvg5JV_6PZaQ'
```
Comment (lptaylor) [inline app/views/ev_map/show.html.erb]: Need to pull this token out of the view for sure before we get popped for it. We have a way to bring it in per Mary

File: app/models/place.rb [hunk 2]
```diff
@@ -8,9 +8,10 @@ class Place
               :coordinates,
               :location,
               :display_phone,
-              :distance
+              :distance,
+              :category
 
-  def initialize(attributes)
+  def initialize(attributes, category)
     @name = attributes[:name]
```
Comment (lptaylor) [inline app/models/place.rb]: Nice thinking ahead so we can put specific markers for the place.

File: app/views/map/show.html.erb [hunk 3]
```diff
@@ -13,5 +13,28 @@
     const <%= "marker#{i}" %> = L.marker([<%= biz.coordinates[:latitude] %>, <%= biz.coordinates[:longitude] %>]).addTo(mymap);
     <%= "marker#{i}" %>.bindPopup("<%= biz.popup_html.html_safe %>").openPopup();
   <% end %>
+
+  function chooseLocal(event) {
+    event.preventDefault();
+    const url = `${event.currentTarget.origin}/ev_search`;
```
Comment (Bradniedt) [inline app/views/map/show.html.erb]: Love that this lets us use both local host and production domain!

File: app/views/map/show.html.erb [hunk 4]
```diff
@@ -13,5 +13,28 @@
...
+
+    fetch(url, {
+           method: "POST",
+           headers: {
+               "Content-Type": "application/json",
+           },
+           redirect: "follow",
+           body: JSON.stringify(data)
+         }).then(function(response) {
+           window.location.replace(response.url);
```
Comment (Bradniedt) [inline app/views/map/show.html.erb]: Nice find!

**Pull request merged**